vendor: update github.com/opencontainers/runc

giuseppe opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:

/kind bug

**What this PR does / why we need it**:

when the systemd cgroup manager is used, controllers not handled by systemd are created manually afterwards.

libcontainer didn't correctly cleanup these cgroups that were leakedon cgroup v1.
    
**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #92766

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

giuseppe committed:
kubelet: skip setting the devices cgroup

use the new libcontainer feature of skipping setting the devices
cgroup.  This is necessary on cgroup v2 to avoid leaking a eBPF
program every time the cgroup is re-configured.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

giuseppe commented:
/sig-node

liggitt commented:
vendor mechanics lgtm

cc @kubernetes/sig-node-pr-reviews for runc and ebpf changes

sjenning commented:
/assign

sjenning commented:
/lgtm

needs top level approver

sjenning commented:
/assign @liggitt

liggitt commented:
there's a gofmt error that needs fixing

sjenning commented:
ah yes https://github.com/kubernetes/kubernetes/pull/92862/commits/52bf6e52d27a88a75562a4563549e76ebbf65530#diff-20728a5825a9e01d8bf61ad43ab2b7e4R128
```
Verifying verify-gofmt.sh

+++ Running case: verify.gofmt 
+++ working dir: /home/prow/go/src/k8s.io/kubernetes
+++ command: bash "hack/make-rules/../../hack/verify-gofmt.sh"
diff -u ./pkg/kubelet/dockershim/cm/container_manager_linux.go.orig ./pkg/kubelet/dockershim/cm/container_manager_linux.go
--- ./pkg/kubelet/dockershim/cm/container_manager_linux.go.orig	2020-07-07 15:02:04.734181675 +0000
+++ ./pkg/kubelet/dockershim/cm/container_manager_linux.go	2020-07-07 15:02:04.734181675 +0000
@@ -123,8 +123,8 @@
 		Parent: "/",
 		Name:   name,
 		Resources: &configs.Resources{
-			Memory:     int64(memoryLimit),
-			MemorySwap: -1,
+			Memory:      int64(memoryLimit),
+			MemorySwap:  -1,
 			SkipDevices: true,
 			Devices: []*configs.DeviceRule{
 				{

Run ./hack/update-gofmt.sh
+++ exit code: 1
+++ error: 1
[0;31mFAILED[0m   verify-gofmt.sh	24s
```

mrunalp commented:
/lgtm

liggitt commented:
/approve
/retest

kolyshkin reviewed (approved):
LGTM (not a maintainer)

kolyshkin commented on pkg/kubelet/dockershim/cm/container_manager_linux.go:
```diff
@@ -123,8 +123,9 @@ func createCgroupManager(name string) (cgroups.Manager, error) {
+			Memory:      int64(memoryLimit),
+			MemorySwap:  -1,
+			SkipDevices: true,
 			Devices: []*configs.DeviceRule{
```
@giuseppe I think that `Devices: ` entry can be `nil` now, so makes sense to remove it from here to avoid confusion.

kolyshkin replied:
same nit in the other two places.

giuseppe replied:
it seems to be still used to setup the systemd service.  If I drop the `Devices` entry then all devices are blocked.

kolyshkin replied:
Ah, that is a bug in runc :( I'll follow up on that one.

kolyshkin replied:
Better late than never, https://github.com/opencontainers/runc/pull/2958 🤡

haircommander commented:
/retest

haircommander commented:
hm is prow stalled or is 24 hours waiting for it to retest normal

giuseppe committed:
vendor: update github.com/opencontainers/runc

when the systemd cgroup manager is used, controllers not handled by
systemd are created manually afterwards.
libcontainer didn't correctly cleanup these cgroups that were leaked
on cgroup v1.

Closes: https://github.com/kubernetes/kubernetes/issues/92766

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

giuseppe commented:
had to rebase.  Pushed a new version

liggitt commented:
/lgtm

liggitt commented:
/retest

haircommander commented:
/retest

haircommander commented:
/retest
